example/workflow/extension/src/workflow-editor-provider.ts

```diff
@@ -0,0 +1,172 @@
+/********************************************************************************
+ * Copyright (c) 2021 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.01
+ ********************************************************************************/
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+
+import { isActionMessage, isWebviewReadyMessage } from 'sprotty-vscode-protocol';
+import { GlspVscodeAdapter } from '@eclipse-glsp/vscode-integration';
+
+const DIAGRAM_TYPE = 'workflow-diagram';
+
+export default class WorkflowEditorProvider implements vscode.CustomEditorProvider {
```

Comment:
From what I can see most of the code in `WorkflowEditorProvider` is generic and will be most likely reused as is by the majority of implementors.
I'd suggest to extract a common, abstract reusable base class (`GlspEditorProvider`). This base class should probably provide a abstract `diagramType` property
and a protected method to generate the intial html webview code ( Which is currently directly set at the end of the `resolveCustomEditor()` method)

Reply:
I extracted a base class (`GlspEditorProvider`) to be used as `CustomEditor` like you described: 01d288dce7bbc09f138fd9afbb1e6fbc11bb3652

I chose to define the method used to generate the webview code as `abstract`. I didn't like the feeling of making assumptions about where users have their scripts for the webview.

Nevertheless, the amount of needed boilerplate code shrinked a lot and the implementation needed for the setup function is trivial. Users can always refer to the example for inspiration if needed.

Reply:
:+1: